core/io/tcp_server.cpp

```diff
@@ -43,7 +43,7 @@ Error TCPServer::listen(uint16_t p_port, const IPAddress &p_bind_address) {
 	ERR_FAIL_COND_V(!_sock.is_valid(), ERR_UNAVAILABLE);
 	ERR_FAIL_COND_V(_sock->is_open(), ERR_ALREADY_IN_USE);
 	ERR_FAIL_COND_V(!p_bind_address.is_valid() && !p_bind_address.is_wildcard(), ERR_INVALID_PARAMETER);
-	ERR_FAIL_COND_V_MSG(p_port < 0 || p_port > 65535, ERR_INVALID_PARAMETER, "The local port number must be between 0 and 65535 (inclusive).");
+	// Do not check if the port range exceeds uint16_t.
```

Comment:
I wouldn't put a comment to day what the code doesn't do, that doesn't seem useful.

Now here the problem might be that since scripting langs are int64_t, users might pass parameters out of bounds, but it will wrap around anyway before the check since the method takes uint16_t. So not much we can do aside from exposing a method with int64_t instead (poke @Faless).

Reply:
I resolved the change by removing the comment, but leaving this comment here until Faless replies.